From d40e640873870eda3b35fe3aa72a0e7dde65b7eb Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Fri, 22 Dec 2023 09:32:39 +0100
Subject: [PATCH 49/N] fixup! path-conversion: Introduce ability to switch off
 conversion.

It would just remove the argument instead of leaving it alone
---
 winsup/cygwin/msys2_path_conv.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/msys2_path_conv.cc b/winsup/cygwin/msys2_path_conv.cc
index 1e9cdbe..f6e91a2 100644
--- a/winsup/cygwin/msys2_path_conv.cc
+++ b/winsup/cygwin/msys2_path_conv.cc
@@ -346,8 +346,10 @@ path_type find_path_start_and_type(const char** src, int recurse, const char* en
      */
     const char *no_pathconv = getenv ("MSYS_NO_PATHCONV");
 
-    if (no_pathconv)
-      return NONE;
+    if (no_pathconv) {
+        *src = end;
+        return NONE;
+    }
 
     /* Let's not convert ~/.file to ~C:\msys64\.file */
     if (*it == '~') {
